src/sedml/SedBase.cpp

```diff
@@ -281,13 +293,20 @@ SedBase::getMetaId ()
 const string&
 SedBase::getId() const
 {
-  return mId;
+    if (!mIdAllowedPreV4 && getVersion() < 4 && getLevel() == 1) {
+        static string empty;
+        return empty;
+    }
+    return mId;
 }
 
 const std::string& SedBase::getName() const
 {
-  static string empty;
-  return empty;
+    if (!mNameAllowedPreV4 && getVersion() < 4 && getLevel() == 1) {
```

Comment:
is this really necessary? I'd let people return the member variable in any case.

Reply:
In theory, I'd be OK with letting people get it?  The issue is more that 'getName' (and 'getId') is used internally a lot, and I'm hesitant to tell that code that an attribute is set if it's not going to be written out.  (For instance, 'isSetName' calls 'getName' and checks if that's empty.)

If people need to get the name, they can always just set the level to 4, and then 'getName' will work if there's something there.